src/app/content/search/hooks.spec.ts

```diff
@@ -93,132 +108,84 @@ describe('hooks', () => {
 
-      expect(dispatch).not.toHaveBeenCalled();
-    });
+      store.dispatch(navigationLocationChange({
+        location: { hash: 'elementId', search: `?query=asdf&target=${mockSearchScrollTarget}` },
+      } as any));
 
-    it('doesn\'t dispatch for PUSH actions', () => {
-      hook({
-        action: 'PUSH',
-        location: {
-          ...assertWindow().location,
-          state: {},
-        },
-        match: {} as any,
-      });
-      hook({
-        action: 'PUSH',
-        location: {
-          ...assertWindow().location,
-          state: {
-            search: 'asdf',
-          },
-        },
-        match: {} as any,
-      });
+      hook({} as any);
 
-      expect(dispatch).not.toHaveBeenCalled();
+      expect(dispatch).toHaveBeenCalledWith(requestSearch('asdf', { isResultReload: true, searchScrollTarget }));
     });
 
-    it('searches for saved query on POP if it is differet from current query', () => {
-      store.dispatch(requestSearch('asdf'));
+    it('searches if there is only search query in the URL', () => {
+      store.dispatch(navigationLocationChange({
+        location: { hash: 'elementId', search: `?query=asdf` },
+      } as any));
 
-      hook({
-        action: 'POP',
-        location: {
-          ...assertWindow().location,
-          state: { search: {query: 'qwer'} },
-        },
-        match: {} as any,
-      });
+      hook({} as any);
 
-      expect(dispatch).toHaveBeenCalledWith(requestSearch('qwer'));
+      expect(dispatch).toHaveBeenCalledWith(requestSearch('asdf'));
     });
 
-    it('searches for saved query with selectedResult on POP if it is differet from current query', () => {
-      store.dispatch(requestSearch('asdf'));
+    it('noops if search query is the same', () => {
+      store.dispatch(navigationLocationChange({
+        location: { hash: 'elementId', search: `?query=asdf` },
+      } as any));
 
-      const hit = makeSearchResultHit({book, page});
-      const selectedResult = {result: hit, highlight: 0};
-      hook({
-        action: 'POP',
-        location: {
-          ...assertWindow().location,
-          state: { search: {query: 'qwer', selectedResult}},
-        },
-        match: {} as any,
-      });
+      hook({} as any);
+
+      dispatch.mockClear();
 
-      expect(dispatch).toHaveBeenCalledWith(requestSearch('qwer', {isResultReload: true, selectedResult}));
+      hook({} as any);
+
+      expect(dispatch).not.toHaveBeenCalledWith();
     });
 
-    it('dispatches selectSearchResult if query and page are the same', () => {
+    it('selects search result', () => {
+      const hit = makeSearchResultHit({book, page});
+      const searchScrollTarget = { type: 'search', index: 0 } as SearchScrollTarget;
+      store.dispatch(navigationLocationChange({
+        location: { hash: hit.source.elementId, search: `?query=asdf&target=${JSON.stringify(searchScrollTarget)}` },
```

Comment:
This is the same as `mockSearchScrollTarget`, right? Maybe consider making both stringified and normal versions of it available in `syncSearch` tests